test/sequential/bootstrap.claimAndActivate.test.ts

```diff
@@ -0,0 +1,299 @@
+
+  const liquidityID = await getLiquidityAssetId(
+    MGA_ASSET_ID,
+    bootstrapCurrency
+  );
+
+  if (promotedPool === true) {
+    const promotingPool = await promotePool(sudo.keyRingPair, liquidityID);
+    expect(promotingPool.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  }
+
+  const claimAndActivate1 = await claimAndActivateBootstrap(testUser1);
+  eventResponse = getEventResultFromMangataTx(claimAndActivate1);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  const claimAndActivate2 = await claimAndActivateBootstrap(testUser2);
+  eventResponse = getEventResultFromMangataTx(claimAndActivate2);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  // finalaze bootstrap
+  const bootstrapFinalize = await finalizeBootstrap(sudo);
+  eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  const bootstrapUser1Liquidity = await api.query.tokens.accounts(
+    testUser1.keyRingPair.address,
+    liquidityID
+  );
+
+  const bootstrapUser2Liquidity = await api.query.tokens.accounts(
+    testUser2.keyRingPair.address,
+    liquidityID
+  );
+
+  return {
+    bootstrapUser1Liquidity: bootstrapUser1Liquidity,
+    bootstrapUser2Liquidity: bootstrapUser2Liquidity,
+    bootstrapExpectedUserLiquidity: bootstrapExpectedUserLiquidity,
+  };
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  keyring = new Keyring({ type: "sr25519" });
+
+  sudo = new User(keyring, sudoUserName);
+  keyring.addPair(sudo.keyRingPair);
+});
+
+describe.each`
+  promoting | vesting
+  ${true}   | ${false}
+  ${false}  | ${false}
+  ${false}  | ${true}
+`(
+  "bootstrap - checking bootstrapped pool in different situations",
+  ({ promoting, vesting }) => {
+    beforeEach(async () => {
+      const api = getApi();
+      bootstrapPhase = await api.query.bootstrap.phase();
+      if (bootstrapPhase.toString() === "Finished") {
+        const bootstrapFinalize = await finalizeBootstrap(sudo);
+        eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      }
+
+      bootstrapPhase = await api.query.bootstrap.phase();
+      expect(bootstrapPhase.toString()).toEqual("BeforeStart");
+
+      const bootstrapCurrencyIssue = await sudoIssueAsset(
+        sudo.keyRingPair,
+        toBN("1", 20),
+        sudo.keyRingPair.address
+      );
+      const bootstrapEventResult = await getEventResultFromMangataTx(
+        bootstrapCurrencyIssue,
+        ["tokens", "Issued", sudo.keyRingPair.address]
+      );
+      const bootstrapAssetId = bootstrapEventResult.data[0].split(",").join("");
+      bootstrapCurrency = new BN(bootstrapAssetId);
+      // bootstrapCurrency = await getNextAssetId();
+      //testUser1.addAsset(bootstrapCurrency);
+
+      [testUser1, testUser2] = setupUsers();
```

Comment:
We need to use this function to set up User. It has EQUAL code for sudoUser and User that we used, but if I write it here as text, it doesn't work. I didn't understand why.

Reply:
maybe some missing initialization for API or something.  If works, leave this like it is for now :)